packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/getFileEntityIdWithSameName.ts

```diff
@@ -0,0 +1,54 @@
+import { SynapseClient } from '@sage-bionetworks/synapse-client'
+import { allowNotFoundError } from '../../../synapse-client/SynapseClientUtils'
+import {
+  convertToEntityType,
+  entityTypeToFriendlyName,
+} from '../../functions/EntityTypeUtils'
+
+/**
+ * Gets the ID of the file entity with the given name whose parent has the given ID.
+ *
+ * @param name The name of the entity to find.
+ * @param parentId The ID of the parent that the found entity must have.
+ * @return The ID of the file entity with the given name and parent ID, or null if the entity does not exist
+ * @throws Error If an entity with given name and parent ID was found, but that entity
+ *         was not a File Entity.
+ */
+export async function getFileEntityIdWithSameName(
```

Comment:
In the uploader, this function will tell us if we should ask the user if a file should be uploaded as a new version of an existing entity.